clayui.com/src/utils/localStorage.tsx

```diff
@@ -0,0 +1,35 @@
+/**
+ * SPDX-FileCopyrightText: © 2020 Liferay, Inc. <https://liferay.com>
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+const isSupported = () => {
+	try {
+		window.localStorage.setItem('_TEST_KEY_', '_TEST_KEY_');
+
+		return true;
+	} catch {
+		return false;
+	}
+};
+
+const getItem = (name: string): string | null => {
+	if (isSupported()) {
+		return window.localStorage.getItem(name);
+	}
+
+	return null;
+};
+
+const setItem = (name: string, value: string): void => {
+	if (isSupported()) {
+		window.localStorage.setItem(name, value);
+	}
+};
+
+const localStorage = {
+	getItem,
+	setItem,
+};
+
+export default localStorage;
```

Comment:
Nice and simple.

![kylo-approves](https://user-images.githubusercontent.com/7074/90494628-fb190200-e143-11ea-995c-7b807ae6ed42.gif)

> One thing I'm not certain about is the structure of the function, I made it similar to how we make our components, defining low-level `getItem` and `setItem` methods and then combining them into a high-level `localStorage` function.

This seems fine to me:

- Use of `default` export makes it easy to consume.
- Rarely use `getItem` without a matching `setItem` partner, so no need to keep them separate (for tree-shaking).
- Exporting single object matches the API that you're wrapping (ie. `window.localStorage` is a single object too).